File: AppJAVA/app/src/main/java/com/example/appjava/MainActivity.java
package com.example.appjava;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.os.Bundle;
import android.os.PowerManager;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

     public void powerService(){
        PowerManager powerManager = (PowerManager) getApplicationContext().getSystemService(Context.POWER_SERVICE);
     }
}

File: View/app/src/main/java/com/example/application/MainActivity.kt
package com.example.application

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import android.net.wifi.WifiManager
import android.os.Bundle

class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onResume() {
        super.onResume()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

    }
    // 检查设备是否安装有该组件
    fun checkActivityEmpety(intent: Intent,flag: Int): Boolean{
        var packageManager = application.packageManager
        if (packageManager.resolveActivity(intent,PackageManager.MATCH_DEFAULT_ONLY) == null){
            return false
        }
        return true
    }

    fun checkServiceEmpety(intent: Intent,flag: Int): Boolean{
        var packageManager = application.packageManager
        if (packageManager.resolveService(intent,PackageManager.MATCH_DEFAULT_ONLY) == null){
            return false
        }
        return true
    }
    
}
